core/main_test.go

```diff
@@ -0,0 +1,22 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package core
+
+import (
+	"testing"
+
+	"go.uber.org/goleak"
+)
+
+// TestMain uses goleak to verify tests in this package do not leak unexpected
+// goroutines.
+func TestMain(m *testing.M) {
+	opts := []goleak.Option{
+		// No good way to shut down these goroutines:
+		goleak.IgnoreTopFunction("github.com/ava-labs/coreth/core/state/snapshot.(*diskLayer).generate"),
+		goleak.IgnoreTopFunction("github.com/ava-labs/coreth/metrics.(*meterArbiter).tick"),
+		goleak.IgnoreTopFunction("github.com/syndtr/goleveldb/leveldb.(*DB).mpoolDrain"),
```

Comment:
q: why not?

Reply:
There is no good way to wait for finishing of a goroutine that is started by code that doesn't track it with a wg or expose the wg to the public interface.
Feel free to open a PR and fix if you find a way :)